edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs

```diff
@@ -17,17 +18,20 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes
 
     public class KubernetesRuntimeInfoProvider : IRuntimeInfoProvider, IRuntimeInfoSource
     {
+        static readonly TimeSpan SystemInfoTimeout = TimeSpan.FromSeconds(3);
         readonly string deviceNamespace;
         readonly IKubernetes client;
         readonly ConcurrentDictionary<string, ModuleRuntimeInfo> moduleRuntimeInfo;
         readonly IModuleManager moduleManager;
+        readonly CancellationTokenSource tokenSource;
 
         public KubernetesRuntimeInfoProvider(string deviceNamespace, IKubernetes client, IModuleManager moduleManager)
         {
             this.deviceNamespace = Preconditions.CheckNonWhiteSpace(deviceNamespace, nameof(deviceNamespace));
             this.client = Preconditions.CheckNotNull(client, nameof(client));
             this.moduleManager = Preconditions.CheckNotNull(moduleManager, nameof(moduleManager));
             this.moduleRuntimeInfo = new ConcurrentDictionary<string, ModuleRuntimeInfo>();
+            this.tokenSource = new CancellationTokenSource(SystemInfoTimeout);
```

Comment:
As discussed, this is due to the sequence of sidecarproxy and edgeagent running time. It is similar to envoy sidecar with applicaiton container. Long term, we can try the pod readiness-gate https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate